File: app/src/main/java/io/moonshard/moonshard/models/roomEntities/ChatEntity.java
package io.moonshard.moonshard.models.roomEntities;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import java.util.ArrayList;
import io.moonshard.moonshard.models.jabber.GenericUser;

@Entity(tableName = "chats")
public class ChatEntity {
    @PrimaryKey
    @NonNull
    public String jid;
    @ColumnInfo
    public String chatName;
    @ColumnInfo
    public ArrayList<GenericUser> users;
    @ColumnInfo
    public int unreadMessagesCount;
    @ColumnInfo
    public String firstMessageUid;
    @ColumnInfo
    public Boolean isGroupChat;

    public ChatEntity(@NonNull String jid, String chatName, ArrayList<GenericUser> users,
                      int unreadMessagesCount, String firstMessageUid,Boolean isGroupChat) {
        this.jid = jid;
        this.chatName = chatName;
        this.users = users;
        this.unreadMessagesCount = unreadMessagesCount;
        this.firstMessageUid = firstMessageUid;
        this.isGroupChat = isGroupChat;
    }

    public boolean isPrivateChat() {
        return users.size() == 2;
    }
}

File: app/src/main/java/io/moonshard/moonshard/models/dbEntities/ChatEntity.kt
package io.moonshard.moonshard.models.dbEntities

import io.moonshard.moonshard.models.api.RoomPin
import io.objectbox.annotation.Backlink
import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id
import io.objectbox.annotation.Unique
import io.objectbox.relation.ToMany

@Entity
data class ChatEntity(
    @Id var id: Long = 0,
    @Unique var jid: String,
    var chatName: String = "",
    var isGroupChat: Boolean = false,
    var unreadMessagesCount: Int = 0
) {
    var event: RoomPin?=null
    var address:String? = null
    var description:String? = null


    lateinit var users: ToMany<ChatUser>

    @Backlink(to = "chat")
    lateinit var messages: ToMany<MessageEntity>
}
